api/src/opentrons/protocol_api/core/engine/instrument.py

```
@@ -2013,6 +1818,95 @@ def _get_location_and_well_core_from_next_tip_info(
             tip_well,
         )
 
+    def _get_transfer_properties_for_tip_rack(
+        self, liquid_class: LiquidClass, tip_rack_uri: str
+    ) -> TransferProperties:
+        try:
+            return liquid_class.get_for(
+                pipette=self.get_pipette_name(), tip_rack=tip_rack_uri
+            )
+        except NoLiquidClassPropertyError:
+            if self._protocol_core.robot_type == "OT-2 Standard":
```

Would it be clearer to move this `if robot_type == "OT-2 Standard"` up above the `try`? Or do you always want to attempt the lookup even if it's an OT-2?

---

Wait, I guess you do always want to try, because PD generates a liquid class for the OT-2.

---

Although the error message is a bit misleading for the OT-2 then?

I.e., say we *are* on the OT-2, and the user has specified a custom liquid class for the pipette, but they gave the wrong name for the tiprack they're using. This function would fail with the error message `Default liquid classes are not supported with OT-2 pipettes and tip racks`, which is less helpful than the default message of `No properties found for {tiprack_uri} for {pipette_name}`.

---

Hmm, yeah I see what you're saying. There's two things I wanted to accomplish with this error message (which was added in an earlier PR): make it clear that no OT-2 pipette and tip rack was supported by the default liquid classes (rather than saying a certain pipette and tip rack were unavailable, which many give someone the idea that other OT-2 combinations may be possible), while also allowing for it if the user added it in some way (editing json, PD, custom liquid classes, etc). But you're right in that the existing error message gets in the way of the second case if the user provides the wrong pipette/tip rack. I'm not sure what the right solution is. Maybe we raise a different error for a wrong tip rack when a pipette exists? Check if any OT-2 compatible pipettes and tip racks exist?

---

That's a good point @ddcc4 
I think we should update the error message to give the full explanation. Something like "No properties found for the pipette and tiprack to be used for this transfer. Note that in-built Opentrons liquid classes do not include properties for the OT-2 by default. If you need to use liquid classes based transfers on the OT-2 then first create a custom liquid class using `define_liquid_class()`, and then specify this custom liquid class in the transfer function."